controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -222,11 +222,22 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.
 			reterr = kerrors.NewAggregate([]error{reterr, err})
 		}
 
-		// TODO: remove this as soon as we have a proper remote cluster cache in place.
-		// Make KCP to requeue in case status is not ready, so we can check for node status without waiting for a full resync (by default 10 minutes).
-		// Only requeue if we are not going in exponential backoff due to error, or if we are not already re-queueing, or if the object has a deletion timestamp.
-		if reterr == nil && !res.Requeue && res.RequeueAfter <= 0 && kcp.ObjectMeta.DeletionTimestamp.IsZero() {
-			if !kcp.Status.Ready {
+		if reterr == nil && kcp.ObjectMeta.DeletionTimestamp.IsZero() {
+			// TODO: remove this as soon as we have a proper remote cluster cache in place.
+			// Make KCP to requeue in case status is not ready, so we can check for node status without waiting for a full resync (by default 10 minutes).
+			// Only requeue if we are not going in exponential backoff due to error, or if we are not already re-queueing, or if the object has a deletion timestamp.
```

Comment:
```suggestion
			// Make KCP requeue in case node status is not ready, so we can check for node status without waiting for a full resync (by default 10 minutes). Only requeue if there is no error, Requeue or RequeueAfter and the object does not have a deletion timestamp.
			// The alternative solution would be to watch the control plane nodes in the Cluster - similar to how the MachineSet and MachineHealthCheck controllers watch the nodes under their control.
```